src/java.desktop/macosx/native/libosxapp/JNIUtilities.h

```diff
@@ -29,11 +29,24 @@
 #include "jni.h"
 #include "jni_util.h"
 
+#import <Cocoa/Cocoa.h>
+
 /********        LOGGING SUPPORT    *********/
 
 #define LOG_NULL(dst_var, name) \
    if (dst_var == NULL) { \
        NSLog(@"Bad JNI lookup %s\n", name); \
+       NSLog(@"%@",[NSThread callStackSymbols]); \
+       if ([NSThread isMainThread] == NO) { \
+           if ((*env)->ExceptionOccurred(env) == NULL) { \
+              JNU_ThrowInternalError(env, "Bad JNI Lookup"); \
+           } else { \
+              if ((*env)->ExceptionOccurred(env) != NULL) { \
+                  (*env)->ExceptionDescribe(env); \
+              } \
```

Comment:
So the update here is that if we are not on the appkit thread, make sure a java exception is thrown.
If we are on the appkit thread, clear any java exception since it isn't going anywhere but do it using
describe which prints it.

Reply:
I have a paren in the wrong place ! I've pushed an update.